### PR TITLE
feature(Stock Received and Stock Purchased)

### LINKED ISSUE #3988: Stock Report : stock received versus stock purchased report
BHIMA supports partial stock entry, so with this issue we must be able to show the difference between what we purchase with what we have entered in the stock.

This report must have, for a purchase : 
- Inventory code
- Inventory description
- purchased quantity
- entry quantity
- quantity difference
- last entry date

### PR DESCRIPTION
**Pull request by lomamech**
- Implementation of the report allowing analysis of the stocks ordered
  from the stocks received

- Implementing bhPurchaseOrderSelect

closes #3988

### CODE REVIEW

**Comment by kwilu** [comment]
Hello @lomamech, it seems like you have referenced #3988 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3988.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by lomamech** [comment]
@mbayopanda  this PR is ready for a review

**Comment by lomamech** [comment]
@mbayopanda  this PR is ready for a review

**Comment by lomamech** [comment]
@mbayopanda  this PR is ready for a review

File: client/src/i18n/fr/tree.json [hunk 1]
```diff
@@ -107,6 +107,7 @@
     "PROJECT":"Projets",
     "PURCHASING":"Achat",
     "PURCHASE_ORDER":"Achats",
+    "PURCHASE_ORDER_ANALYSIS": "[Stock] Analyse de commande d'achat",
```
Comment (mbayopanda) [inline client/src/i18n/fr/tree.json:112]: 👍

File: server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                  <td colspan="8"></td>
+                </tr>
+                <tr>
+                  <td class="text-uppercase text-right text-bold" colspan="6">
+                    {{translate 'FORM.LABELS.TOTAL'}}
+                  </td>
+                  <td class="text-bold" colspan="2"> {{ quantity_inStock }} </td>
+                </tr>
+                <tr>
+                  <td class="text-uppercase text-right text-bold" colspan="6">
```
Comment (mbayopanda) [inline server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars]: You can also remove this `text-uppercase` class.

File: server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                    {{translate 'FORM.LABELS.QUANTITY_ORDERED'}}
+                  </td>
+                  <td class="text-bold" colspan="2"> {{ quantity_ordered }} </td>
+                </tr>
+                <tr>
+                  <td class="text-uppercase text-right text-bold" colspan="6">
```
Comment (mbayopanda) [inline server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars]: You can also remove this `text-uppercase` class.

File: server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    {{translate 'FORM.LABELS.QUANTITY_DIFFERENCE'}}
+                  </td>
+                  <td class="text-bold" colspan="2"> {{ quantity_difference }} </td>
+                </tr>
+                <tr>
+                  <td class="text-uppercase text-right text-bold" colspan="6">
```
Comment (mbayopanda) [inline server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars]: You can also remove this `text-uppercase` class.

File: server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+          <br>
+          <table style="width: 90%; margin-right: auto; margin-left: auto" class="table table-striped table-condensed table-report table-bordered">
+            <thead>
+              <th width="6%" class="text-uppercase"> {{translate 'FORM.LABELS.CODE'}} </th>
+              <th width="20%" class="text-uppercase"> {{translate 'FORM.LABELS.INVENTORY'}} </th>
+              <th width="12%" class="text-uppercase"> {{translate 'FORM.LABELS.QUANTITY_ORDERED'}} </th>
+              <th width="12%" class="text-uppercase"> {{translate 'FORM.LABELS.QUANTITY_ENTRY'}} </th>
+              <th width="15%" class="text-uppercase"> {{translate 'FORM.LABELS.QUANTITY_DIFFERENCE'}} </th>
+              <th width="17%" class="text-uppercase" colspan="2"> {{translate 'FORM.LABELS.STATUS'}} </th>
+              <th width="18%" class="text-uppercase"> {{translate 'FORM.LABELS.LAST_ENTRY_DATE'}} </th>
```
Comment (mbayopanda) [inline server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars:54]: Could you remove all these `text-uppercase` classes.

File: package.json [hunk 6]
```diff
@@ -8,9 +8,9 @@
     "dev": "cross-env NODE_ENV=development gulp build && cd bin && cross-env NODE_ENV=development node server/app.js",
     "test": "./sh/test.sh",
     "test:integration": "cross-env SUITE=\"integration\" ./sh/test.sh",
-    "test:ends": "cross-env SUITE=\"end-to-end\" ./sh/test.sh",
-    "test:client-unit": "cross-env SUITE=\"client-unit\" ./sh/test.sh",
-    "test:server-unit": "cross-env SUITE=\"server-unit\" ./sh/test.sh",
+    "test:ends": "SUITE=\"end-to-end\" ./sh/test.sh",
+    "test:client-unit": "SUITE=\"client-unit\" ./sh/test.sh",
+    "test:server-unit": "SUITE=\"server-unit\" ./sh/test.sh",
```
Comment (mbayopanda) [inline package.json]: Could you restore `cross-env` command, such as : 
```json
"test:ends": "cross-env SUITE=\"end-to-end\" ./sh/test.sh",
"test:client-unit": "cross-env SUITE=\"client-unit\" ./sh/test.sh",
"test:server-unit": "cross-env SUITE=\"server-unit\" ./sh/test.sh",
```

**Review by mbayopanda — COMMENTED** [review]
@lomamech this is a good job, the report is very nice.

I made just some comments.

**Comment by kwilu** [comment]
Hello @lomamech, it seems like you have referenced #3988 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3988.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by lomamech** [comment]
@mbayopanda  all things is Okay

**Comment by kwilu** [comment]
Hello @lomamech, it seems like you have referenced #3988 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3988.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by lomamech** [comment]
@mbayopanda i fix the conflict

**Pull request merged**